tests/baselines/reference/tsbuild/amdModulesWithOut/prepend-reports-deprecation-error.js

```diff
@@ -447,6 +447,20 @@ declare const globalConst = 10;
       "./file2.ts": "-13729954175-export const y = 20;",
       "./global.ts": "1028229885-const globalConst = 10;"
     },
+    "root": [
+      [
+        [
+          2,
+          5
+        ],
+        [
+          "./file0.ts",
+          "./file1.ts",
+          "./file2.ts",
+          "./global.ts"
+        ]
```

Comment:
My understanding is that this encoding is either a range of indexes over some array, or if noncontiguous, the filename from that array. Would it be more efficient to instead store either ranges or indexes, like:

```ts
[[2, 5], 7, 9]
```

Or, am I misinterpreting the format?

Reply:
It is stored as ranges or indices. 

So here are examples: 
 `[[2,5],7,9]` => `2,3,4,5,7,9` eg https://github.com/microsoft/TypeScript/pull/52417/files#diff-5fbf0d47ec6990dcf27581b225204e9671350ee831847ee68d498fddfd852e7dR983
https://github.com/microsoft/TypeScript/pull/52417/files#diff-97195a8f223f06c8bc85c9b5faef6126b946d0a6dc3a47cf3bde7f80e099a444R507

 `[2,3,[7,9]]` => `2,3,7,8,9` eg: https://github.com/microsoft/TypeScript/pull/52417/files#diff-5fbf0d47ec6990dcf27581b225204e9671350ee831847ee68d498fddfd852e7dR603

`[2,[7,9]]` => `2,7,8,9` eg: https://github.com/microsoft/TypeScript/pull/52417/files#diff-5b7eeb5a2a5682fb3f3ddde640382cd676760cd49d3a0513bd1df1fd9c017389R115


And will add some more as a next commit to show these cases clearly

Reply:
[c6d40cb](https://github.com/microsoft/TypeScript/pull/52417/commits/c6d40cb7d62342ada63c4776bec2aad4a997f6b5)

Reply:
Oh, I think my confusion comes from the fact that I was expecting these to not encode filenames at all (to save space), but I guess that's not what's happening.

Reply:
It is not encoding file names.. thats the readable baseline :) where first column is fileId encoded in tsbuildInfo and fileNames it refers to as second member of tuple

Reply:
Look for root in the tsbuildInfo: https://github.com/microsoft/TypeScript/pull/52417/commits/c6d40cb7d62342ada63c4776bec2aad4a997f6b5#diff-6cf9dfab853b1fefb0fdd6192a254278dcd5f9f9e0cdd130503a7b756622473cR68

Reply:
Oh! I thought it was just the same thing pretty printed.